diff --git a/lib/File/GlobMapper.pm b/lib/File/GlobMapper.pm
index 53f957a..6454bc4 100644
--- a/lib/File/GlobMapper.pm
+++ b/lib/File/GlobMapper.pm
@@ -29,6 +29,11 @@ our ($VERSION, @EXPORT_OK);
 $VERSION = '1.001';
 @EXPORT_OK = qw( globmap );
 
+our $BEGIN_DELIM = "\xFF";
+our $END_DELIM   = "\xFE";
+our $BACKSLASH_ESC = "\xFD";
+our $HASH_ESC = "\xFC";
+our $STAR_ESC = "\xFB";
 
 our ($noPreBS, $metachars, $matchMetaRE, %mapping, %wildCount);
 $noPreBS = '(?<!\\\)' ; # no preceding backslash
@@ -310,14 +315,23 @@ sub _parseOutputGlob
     }
 
     my $noPreBS = '(?<!\\\)' ; # no preceding backslash
-    #warn "noPreBS = '$noPreBS'\n";
+    my $noPreESC = '(?<![${BEGIN_DELIM}])' ; # no preceding backslash
 
-    #$string =~ s/${noPreBS}\$(\d)/\${$1}/g;
-    $string =~ s/${noPreBS}#(\d)/\${$1}/g;
-    $string =~ s#${noPreBS}\*#\${inFile}#g;
-    $string = '"' . $string . '"';
+    # escape any use of the delimiter symbols
+    # $string =~ s/(${BEGIN_DELIM}|${END_DELIM}|${BACKSLASH_ESC})/$1$1/g;
+
+    # escape \# and \*
+    $string =~ s/\\#/${HASH_ESC}/g;
+    $string =~ s/\\\*/${STAR_ESC}/g;
+
+    # Transform "#3" to BEGIN_DELIM 3 END_DELIM
+    $string =~ s/${noPreESC}#(\d)/${BEGIN_DELIM}${1}${END_DELIM}/g;
+
+    $string =~ s#\*#${BEGIN_DELIM}${END_DELIM}#g;
+
+    # print "INPUT  '$self->{InputPattern}'\n";
+    # print "OUTPUT '$self->{OutputGlob}' => '$string'\n";
 
-    #print "OUTPUT '$self->{OutputGlob}' => '$string'\n";
     $self->{OutputPattern} = $string ;
 
     return 1 ;
@@ -335,11 +349,31 @@ sub _getFiles
         next if $inFiles{$inFile} ++ ;
 
         my $outFile = $inFile ;
+        my @matches ;
+
+        my $noPreESC = '(?<![${BEGIN_DELIM}])' ; # no preceding backslash
 
-        if ( $inFile =~ m/$self->{InputPattern}/ )
+        if (@matches = ($inFile =~ m/$self->{InputPattern}/ ))
         {
-            no warnings 'uninitialized';
-            eval "\$outFile = $self->{OutputPattern};" ;
+            $outFile = $self->{OutputPattern};
+            my $ix = 1;
+
+            # get the filename glob
+            $outFile =~ s/${noPreESC}${BEGIN_DELIM}${END_DELIM}/$inFile/g;
+
+            # now each of the #1, #2,...
+            for my $pattern (@matches)
+            {
+                $outFile =~ s/${noPreESC}${BEGIN_DELIM}${ix}${END_DELIM}/$pattern/g;
+
+                ++ $ix;
+            }
+
+            # unescape
+            $outFile =~ s/${BEGIN_DELIM}${BEGIN_DELIM}/${BEGIN_DELIM}/g;
+            $outFile =~ s/${END_DELIM}${END_DELIM}/${END_DELIM}/g;
+            $outFile =~ s/${HASH_ESC}/#/g;
+            $outFile =~ s/${STAR_ESC}/*/g;
 
             if (defined $outInMapping{$outFile})
             {
